src/iso9660.rs

```diff
@@ -49,6 +49,19 @@ pub struct IsoFs {
 impl IsoFs {
     pub fn from_file(mut file: fs::File) -> Result<Self> {
         let descriptors = get_volume_descriptors(&mut file)?;
+        match &descriptors[0] {
+            VolumeDescriptor::Boot(_boot_volume_descriptor) => {}
+            VolumeDescriptor::Primary(primary_volume_descriptor) => {
+                if i64::from(primary_volume_descriptor.volume_space_size)
+                    * i64::from(primary_volume_descriptor.logical_block_size)
+                    > file.metadata()?.len().try_into().unwrap()
```

Comment:
Since the `volume_space_size` should be unsigned, these will be `u64` and the `try_into().unwrap()` goes away.